lib/kubernetes-deploy/kubernetes_resource/service.rb

```diff
@@ -41,19 +31,26 @@ def exists?
     private
 
     def exposes_zero_replica_deployment?
-      related_deployment_replicas && related_deployment_replicas == 0
+      @related_deployment_replicas == 0
     end
 
-    def related_deployment_replicas
-      @related_deployment_replicas ||= begin
-        selector = @definition["spec"]["selector"].map { |k, v| "#{k}=#{v}" }.join(",")
-        raw_json, _err, st = kubectl.run("get", "deployments", "--selector=#{selector}", "--output=json")
-        return unless st.success?
+    def selector
+      @selector ||= @definition["spec"]["selector"].map { |k, v| "#{k}=#{v}" }.join(",")
+    end
+
+    def fetch_related_pod_count
+      raw_json, _err, st = kubectl.run("get", "pods", "--selector=#{selector}", "--output=json")
+      return 0 unless st.success?
+      JSON.parse(raw_json)["items"].length
+    end
+
+    def fetch_related_replica_count
+      raw_json, _err, st = kubectl.run("get", "deployments", "--selector=#{selector}", "--output=json")
+      return 1 unless st.success?
```

Comment:
If I understand correctly the point of this is to cause the replica count check to fail if we're unable to retrieve the actual replica count? Should we just use `nil` for that case as opposed to an arbitrary integer not equal to 0?